krux

```diff
@@ -34,7 +34,19 @@ getdevice() {
     for path in `(find /sys/ -name idVendor | rev | cut -d/ -f 2- | rev) 2>/dev/null`; do
         if grep -q $idV $path/idVendor; then
             if grep -q $idP $path/idProduct; then
-                (find $path -name 'device' | rev | cut -d / -f 2 | rev) 2>/dev/null
+                (find $path -name 'device' | rev | cut -d / -f 2 | rev) 2>/dev/null | head -1
```

Comment:
This worked! I just had to replace `num=${2:1}` for `num=${2}`

Reply:
I wonder if there is a way to try both ports, if it fails with `device_num=1` and get the "greeting fail" message, than tries with `device_num=2`. Because I don't know if the naming behavior I get here will repeat elsewhere.